File: backend/apps/graphql/types/github/release.py
from apps.github.models.release import Release
from apps.github.models.user import User
from graphene_django import DjangoObjectType
from apps.graphql.types.github.user import UserType
from graphene import Field


class ReleaseType(DjangoObjectType):
    author = Field(UserType)
    class Meta:
        model = Release
        fields = ('name', 'tag_name', 'is_pre_release', 'published_at', 'author')

File: frontend/src/types/release.ts
import type { RepositoryDetails, User } from 'types/user'

export type Release = {
  __typename?: string
  author?: User | null
  id: string
  isPreRelease?: boolean
  name: string
  organizationName?: string | null
  projectName?: string
  publishedAt: number
  repository?: RepositoryDetails
  repositoryName?: string | null
  tagName: string
  url?: string
}
